cli/src/lib/website/filesInit.ts

```diff
@@ -35,42 +102,24 @@ export async function sendFilesInits(
   metadatas: Metadata[],
   metadatasToDelete: Metadata[]
 ): Promise<Operation[]> {
-  const fileInitsBatches: FileInit[][] = []
-  const operations: Operation[] = []
+  const batches: Batch[] = createBatches(
+    files,
+    filesToDelete,
+    metadatas,
+    metadatasToDelete
+  )
 
-  for (let i = 0; i < files.length; i += batchSize) {
-    fileInitsBatches.push(files.slice(i, i + batchSize))
-  }
+  const operations: Operation[] = []
 
-  for (const batch of fileInitsBatches) {
-    const coins = await filesInitCost(
-      sc,
-      batch,
-      filesToDelete,
-      metadatas,
-      metadatasToDelete
-    )
-    const gas = await estimatePrepareGas(
-      sc,
-      batch,
-      filesToDelete,
-      metadatas,
-      metadatasToDelete
-    )
-    const args = new Args()
-      .addSerializableObjectArray(batch)
-      .addSerializableObjectArray(filesToDelete)
-      .addSerializableObjectArray(metadatas)
-      .addSerializableObjectArray(metadatasToDelete)
-      .serialize()
+  for (const batch of batches) {
+    const coins = await batch.batchCost(sc)
+    const gas = await batch.estimateGas(sc)
+    const args = batch.serialize()
 
     const op = await sc.call(functionName, args, {
-      coins: coins,
+      coins: coins <= 0n ? 0n : coins,
       maxGas: gas,
-      fee:
-        BigInt(gas) > BigInt(Mas.fromString('0.01'))
-          ? BigInt(gas)
-          : BigInt(Mas.fromString('0.01')),
+      fee: gas > Mas.fromString('0.01') ? gas : Mas.fromString('0.01'),
```

Comment:
```Mas.fromString('0.01')``` should be in a const to avoid code duplication

Reply:
In fact, we should even use the minimal fees from the node ! 
Thanks for raising this issue, will fix it